source/game_sa/VisibilityPlugins.h

```diff
@@ -188,4 +188,4 @@ class CVisibilityPlugins {
     static bool VehicleVisibilityCB_BigVehicle(RpClump* clump);
 };
 
-#define RpAtomicGetVisibilityPlugin(atomic) ((tAtomicVisibilityPlugin*)((uint32)atomic + CVisibilityPlugins::ms_atomicPluginOffset))
+#define RpAtomicGetVisibilityPlugin(atomic) RWPLUGINOFFSET(tAtomicVisibilityPlugin, &((RwObject*)(atomic))->flags, CVisibilityPlugins::ms_atomicPluginOffset)
```

Comment:
Okay so now I understand.
`ms_atomicPluginOffset` is set in `CVisibilityPlugins::PluginAttach`. The size of the struct is 4, and so is the size param passed to the plugin create function... So I believe the current code for here would be without the `flags` offset:
```suggestion
#define RpAtomicGetVisibilityPlugin(atomic) RWPLUGINOFFSET(tAtomicVisibilityPlugin, &((RwObject*)(atomic)), CVisibilityPlugins::ms_atomicPluginOffset)
```
Now though you should correct all places that use this function to account for that offset (by using `m_flags` field of `tAtomicVisibilityPlugin`)